Checkout/Service/PaymentMethod.php

```diff
@@ -200,16 +200,20 @@ public function refund(Varien_Object $payment, $amount)
                 . ' Please make sure Bold Checkout module output is enabled and Bold Checkout Integration is on.'
             );
         }
-        $orderGrandTotal = Mage::app()->getStore()->roundPrice($order->getGrandTotal());
-        $amount = Mage::app()->getStore()->roundPrice($amount);
-        if ($orderGrandTotal <= $amount) {
-            $transactionId = Bold_Checkout_Api_Bold_Payment::refundFull($order);
-            $payment->setTransactionId($transactionId)
-                ->setIsTransactionClosed(1)
-                ->setShouldCloseParentTransaction(true);
-            return $this;
+        if(null !== $payment->getCreditmemo()->getTransactionId()){ 
```

Comment:
There should be a space between the `if` statement and the condition as well as between the condition and the opening curly brace.

Yoda conditions should not be used for clarity.